aea/protocols/generator.py

```diff
@@ -118,45 +122,20 @@ def _extract_specification(self) -> None:
 
         :return: None
         """
-        if len(self.yaml_documents) >= 2 and self.yaml_documents[0] is not None and self.yaml_documents[1] is not None:
-            if "name" in self.yaml_documents[0]:
-                self.name = self.yaml_documents[0]["name"]
-                if type(self.name) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'name' is not a string.")
-                if self.name == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'name' is empty.")
-            else:
-                raise ProtocolSpecificationParseError("Protocol name could not be found.")
-            if "authors" in self.yaml_documents[0]:
-                self.authors = self.yaml_documents[0]["authors"]
-                if type(self.authors) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'authors' is not a string.")
-                if self.authors == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'authors' is empty.")
-            if "version" in self.yaml_documents[0]:
-                self.version = self.yaml_documents[0]["version"]
-                if type(self.version) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'version' is not a string.")
-                if self.version == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'version' is empty.")
-            if "license" in self.yaml_documents[0]:
-                self.license_config = self.yaml_documents[0]["license"]
-                if type(self.license_config) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'license' is not a string.")
-                if self.license_config == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'license' is empty.")
-            if "url" in self.yaml_documents[0]:
-                self.url = self.yaml_documents[0]["url"]
-                if type(self.url) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'url' is not a string.")
-                if self.url == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'url' is empty.")
-            if "description" in self.yaml_documents[0]:
-                self.description = self.yaml_documents[0]["description"]
-                if type(self.description) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'description' is not a string.")
-                if self.description == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'description' is empty.")
+        if len(self.yaml_documents) >= 2 \
```

Comment:
It seems like these conditions should be checked at import and otherwise an import error should be raised. Then I would also rename `yaml_documents` to something more descriptive. Since you expect `yaml_documents` to have exactly two items, why not create two variables and assert their correct type on initialisation?

Reply:
Yaml 'document' is the official name of these constructs by yaml.org.

I didn't get your comment about doing the checks on import/initialisation. We need to read the yaml file to check its consistency and that's what the load method and its private sub-methods do.